src/pages/CSSocietyPage.js

```diff
@@ -107,7 +162,18 @@ export default function CSSocietyPage(props) {
           </>
         )}
         <Paper className={classes.paper}>
-          <Typography variant="h3">Executive Committee</Typography>
+          <Typography variant="h3">Executive Committee
+          {loggedIn && (
+            <>
+          <IconButton component="span" onClick={deleteExecom}>
+          <svg xmlns="http://www.w3.org/2000/svg" width="16" height="16" fill="currentColor" class="bi bi-mortarboard" viewBox="0 0 16 16">
+          <path d="M8.211 2.047a.5.5 0 0 0-.422 0l-7.5 3.5a.5.5 0 0 0 .025.917l7.5 3a.5.5 0 0 0 .372 0L14 7.14V13a1 1 0 0 0-1 1v2h3v-2a1 1 0 0 0-1-1V6.739l.686-.275a.5.5 0 0 0 .025-.917l-7.5-3.5ZM8 8.46 1.758 5.965 8 3.052l6.242 2.913L8 8.46Z"/>
+          <path d="M4.176 9.032a.5.5 0 0 0-.656.327l-.5 1.7a.5.5 0 0 0 .294.605l4.5 1.8a.5.5 0 0 0 .372 0l4.5-1.8a.5.5 0 0 0 .294-.605l-.5-1.7a.5.5 0 0 0-.656-.327L8 10.466 4.176 9.032Zm-.068 1.873.22-.748 3.496 1.311a.5.5 0 0 0 .352 0l3.496-1.311.22.748L8 12.46l-3.892-1.556Z"/>
+          </svg>
+          </IconButton>
+          </>
+          )}
+          </Typography>
           <br />
           <Grid container spacing={2} justify="center">
```

Comment:
It doesn't make sense to have this repeating code for each society, let's do one thing, 

- Create a Component called `SocietyExecom` which will take `execom`, `sid` as parameter, we can fetch the `execom` from the parent component and pass it same applies for `sid`.
- Move this 
```javascript
<Grid container spacing={2} justifyContent="center">
  {execom.map((member) => (
    <Grid item xs={12} md={4}>
      <Avatar
        name={member.name}
        position={member.position}
        src={member.image}
      />
    </Grid>
  ))}
</Grid>;

``` 
to `SocietyExecom` so that the execom will be rendered as how it is.

- Inside this add the button for Creating and Ending Execom so that it will be repeated to every society, if you want to be consistent with the placement of the Add Icon, maybe use a `AddIcon` from `'@material-ui/icons/Add'`

```javascript
{
  loggedIn && (
    <>
      <Grid item xs={12} container justifyContent="center" alignItems="center">
        <Tooltip
          title="Add new execom member"
          arrow
          placement="top"
          aria-label="addExecom"
        >
          <span>
            <Fab onClick={handleDialogOpen} aria-label="addExecom">
              <AddIcon className={classes.large} />
            </Fab>
          </span>
        </Tooltip>
      </Grid>
      <AddExecomDialog
        onClose={handleDialogClose}
        aria-label="add-execom-dialog"
        open={dialog}
        sid={ecats.compsoc}
      />
    </>
  );
}

```

- Overall it will look like this 
```javascript
<Grid container spacing={2} justifyContent="center">
  {execom.map((member) => (
    <Grid item xs={12} md={4}>
      <Avatar
        name={member.name}
        position={member.position}
        src={member.image}
      />
    </Grid>
  ))}
  {loggedIn && (
    <>
      <Grid item xs={12} container justifyContent="center" alignItems="center">
        <Tooltip
          title="Add new execom member"
          arrow
          placement="top"
          aria-label="addExecom"
        >
          <span>
            <Fab onClick={handleDialogOpen} aria-label="addExecom">
              <AddIcon className={classes.large} />
            </Fab>
          </span>
        </Tooltip>
      </Grid>
      <AddExecom
        open={dialog}
        onClose={handleDialogClose}
        socname={socname}
        sid={sid}
        execom={execom}
      />
    </>
  )}
</Grid>;
```
Why are even doing this?
 By creating a `SocietyExecom` component and moving the repeated code to it, we can remove **duplication** and improve maintainability. The addition of the Add Icon button for creating and ending execom for each society is a good visual feature, and adding it to the `SocietyExecom` component will make it consistent across all societies.